src/mapped-types.test-d.ts

```diff
@@ -0,0 +1,17 @@
+import { DeepPartial } from './mapped-types';
+
+declare const test: any;
+
+type NestedProps = {
+  first: {
+    second: {
+      name: string;
+    };
+  };
+};
+
+(test as DeepPartial<NestedProps>) = {};
+
+<T extends object>() => {
+  (test as DeepPartial<T>) = {};
+};
```

Comment:
@mpiniarski This approach with all related boilerplate doesn't look manageable, we need something else I consider accepting your suggestion about `tsd` as suggested solution :)

But first would like to understand what's the limitation with `dts`? Wouldn't that code not show error in dts spec file?

Reply:
After closer investigation, it seems to be possible to use `dts`, I've moved the tests to dts spec file.

Reply:
@mpiniarski amazing, thanks for the update :)